pkg/kv/kvserver/client_raft_test.go

```diff
@@ -5572,6 +5572,24 @@ func TestProcessSplitAfterRightHandSideHasBeenRemoved(t *testing.T) {
 			_, store0Exists := rhsInfo.Desc.GetReplicaDescriptor(1)
 			require.True(t, store0Exists)
 
+			// Ensure that store 0 has created the RHS replica in memory from raft
```

Comment:
Is case (1) prone to the same problem? A prefix of the code up to the `waitForTombstone` look identical, except typos:

```diff
> 
26,28c26,28
<                       // Unsuccessful because the RHS will not accept the learner snapshot and
<                       // will be rolled back. Nevertheless, it will have learned that it has
<                       // been removed at the old replica ID.
---
>                       // Unsuccessfuly because the RHS will not accept the learner snapshot
>                       // and will be rolled back. Nevertheless it will have learned that it
>                       // has been removed at the old replica ID.

```